test/src/test/java/org/corfudb/integration/LogReplicationRoutingQueueIT.java

```diff
@@ -0,0 +1,226 @@
+        CorfuRecord<Queue.RoutingTableSnapshotEndMarkerMsg, Message> record = new CorfuRecord<>(endMarker, null);
+
+        Object[] smrArgs = new Object[2];
+        smrArgs[0] = snapshotSyncId;
+        smrArgs[1] = record;
+
+        UUID endMarkerStreamId = CorfuRuntime.getStreamID(TableRegistry.getFullyQualifiedTableName(
+                CORFU_SYSTEM_NAMESPACE, SNAPSHOT_END_MARKER_TABLE_NAME));
+
+        // Add the marker to endMarkerStream
+        try (TxnContext txnContext = corfuStoreSource.txn(CORFU_SYSTEM_NAMESPACE)) {
+            txnContext.logUpdate(endMarkerStreamId, new SMREntry("put", smrArgs,
+                            corfuStoreSource.getRuntime().getSerializers().getSerializer(ProtobufSerializer.PROTOBUF_SERIALIZER_CODE)),
+                    Arrays.asList(CorfuRuntime.getStreamID(String.join("", SNAPSHOT_SYNC_QUEUE_TAG_SENDER_PREFIX, DefaultClusterConfig.getSinkClusterIds().get(0)))));
+            txnContext.commit();
+        } catch (Exception e) {
+            log.error("Failed to add End Marker", e);
+        }
+
+        // Confirm data does exist on Source Cluster
+        try (TxnContext txn = corfuStoreSource.txn(CORFU_SYSTEM_NAMESPACE)) {
+            assertThat(txn.entryList(snapshotQueueSource).size()).isEqualTo(numEntries);
+            txn.commit();
+        } catch (Exception e) {
+            log.info("Exception: ", e);
+        }
+
+        // Confirm data does not exist on Sink Cluster
+        try (TxnContext txn = corfuStoreSink.txn(CORFU_SYSTEM_NAMESPACE)) {
+            Assert.assertTrue(txn.entryList(replicatedQueueSink).isEmpty());
+            txn.commit();
+        } catch (Exception e) {
+            log.info("Exception: ", e);
+        }
+
+        // Subscribe to replication status table on Source to capture the status of snapshot sync completion
+        corfuStoreSink.openTable(CORFU_SYSTEM_NAMESPACE,
+                REPLICATION_STATUS_TABLE_NAME,
+                LogReplicationSession.class,
+                ReplicationStatus.class,
+                null,
+                TableOptions.fromProtoSchema(ReplicationStatus.class));
+
+        CountDownLatch statusUpdateLatch = new CountDownLatch(2);
+        ReplicationStatusListener sinkStatusListener = new ReplicationStatusListener(statusUpdateLatch, false);
+        corfuStoreSink.subscribeListener(sinkStatusListener, CORFU_SYSTEM_NAMESPACE, LR_STATUS_STREAM_TAG);
+
+        // Start LR for both source and sink
+        startSourceLogReplicator();
+        startSinkLogReplicator();
+
+        log.info(">> Wait ... Snapshot log replication in progress ...");
+        statusUpdateLatch.await();
+
+        // Verify data on sink and add records to sink entries set
+        while (true) {
+            try (TxnContext txn = corfuStoreSink.txn(CORFU_SYSTEM_NAMESPACE)) {
+                if (!txn.entryList(replicatedQueueSink).isEmpty()) {
+                    sinkEntries.addAll(txn.entryList(replicatedQueueSink));
+                    break;
```

Comment:
entries may get replicated in multiple transactions so breaking as soon as non-zero entries are found may lead to intermittent failures.  The loop can be modified to check for the expected number of entries